spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/watcher/DependencyStateChangeListenerRegistry.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.zookeeper.discovery.watcher;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.state.ConnectionState;
+import org.apache.curator.x.discovery.ServiceCache;
+import org.apache.curator.x.discovery.details.ServiceCacheListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class DependencyStateChangeListenerRegistry implements ServiceCacheListener {
+
+    private static final Logger log = LoggerFactory.getLogger(DependencyStateChangeListenerRegistry.class);
+
```

Comment:
Hmm... I guess you're right. I'll have to think of how to make it better. Thanks for the suggestion though.